httpbp/headers.go

```diff
@@ -0,0 +1,117 @@
+package httpbp
+
+import (
+	"context"
+	"net/http"
+)
+
+type headerContextKey int
+
+const (
+	// HeaderEdgeContext is the key for the raw edge request context
+	HeaderEdgeContext headerContextKey = iota
+
+	// HeaderSpanTraceID is the header for the trace ID passed by the caller
+	HeaderSpanTraceID
+
+	// HeaderSpanParentID is the header for the parent ID passed by the caller
+	HeaderSpanParentID
+
+	// HeaderSpanID is the header for the span ID passed by the caller
+	HeaderSpanID
+
+	// HeaderSpanFlags is the header for the span flags passed by the caller
+	HeaderSpanFlags
+
+	// HeaderSpanSampled is the header for the sampled flag passed by the caller
+	HeaderSpanSampled
+)
+
+// GetHeader returns the HTTP header stored on the context at key.
+func GetHeader(ctx context.Context, key headerContextKey) (string, bool) {
+	if header, ok := ctx.Value(key).(string); ok {
+		return header, true
+	}
+	return "", false
+}
+
+// HeaderTrustHandler provides an interface PopulateBaseplateRequestContext to
+// verify that it should trust the HTTP headers it receives.
+type HeaderTrustHandler interface {
+	TrustEdgeContext(r *http.Request) bool
+	TrustSpan(r *http.Request) bool
```

Comment:
Can you add more comments into the interface about what do these 2 functions do? I can figure that out by reading the code of `PopulateBaseplateRequestContext`, but that should be part of the documentation :)